contracts/Groth16Verifier.sol

```diff
@@ -20,107 +20,66 @@
 pragma solidity 0.8.23;
 import "./Groth16.sol";
 
-library Pairing {
-  // The prime q in the base field F_q for G1
-  uint private constant _Q =
+contract Groth16Verifier is IGroth16Verifier {
+  uint private constant _P =
```

Comment:
`_P` is `uint` but `_Q` is `uint256`. I'm not sure if this makes any difference, but should be consistent anyway? (personally I would use `uint256` everywhere because that way it's more explicit that these really big numbers; also in `Groth16.sol`)

Reply:
Yes, good catch. They are the same, but `uint256` is indeed preferred.